### PR TITLE
Windows based k8s acceptance tests

### PR DESCRIPTION
**Pull request by aahel**
Changes proposed in this PR:
- refactored acceptance tests to run on windows

How I've tested this PR:
tests from local on windows aks clsuters

How I expect reviewers to test this PR:

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: acceptance/tests/connect/connect_external_servers_test.go
Comment (absolutelightning) [inline acceptance/tests/connect/connect_external_servers_test.go:83]: add a check for enable windows && tproxy -> then skip

File: acceptance/framework/config/config.go
Comment (absolutelightning) [inline acceptance/framework/config/config.go:153]: global. imageConsulDataplaneWindows = global.imageConsuldataplane + ":windows"

File: acceptance/tests/connect/connect_external_servers_test.go
Comment (aahel) [reply acceptance/tests/connect/connect_external_servers_test.go:83]: done

File: acceptance/framework/config/config.go
Comment (aahel) [reply acceptance/framework/config/config.go]: done

**Review by absolutelightning — APPROVED** [review]

File: .github/workflows/build.yml
Comment (Ganeshrockz) [inline .github/workflows/build.yml:245]: The build job itself builds go binaries and upload zip files for various architectures in Windows. Do we need this separate job doing the same on a windows runner?

**Review by Ganeshrockz — COMMENTED** [review]
Awesome job on this PR. Have some minor comments. Looks good otherwise

File: acceptance/framework/config/config.go
Comment (absolutelightning) [reply acceptance/framework/config/config.go:150]: Hey, windows tag is added because @curtbushko had asked us to use one variable and different tags for consulk8simage. We need both images here - linux as well as windows. Hence we added windows tag

File: .github/workflows/build.yml
Comment (absolutelightning) [reply .github/workflows/build.yml:245]: yes. this job has run-on `windows-2019` while others is running on linux machines.

Comment (Ganeshrockz) [reply .github/workflows/build.yml:245]: Why is there a strict requirement to run this on windows? Can't we use the same zip file that got generated from the previous job?

File: acceptance/framework/config/config.go
Comment (Ganeshrockz) [reply acceptance/framework/config/config.go:150]: So the helm key would change in this case?

File: acceptance/tests/cli/cli_upgrade_test.go
Comment (aahel) [reply acceptance/tests/cli/cli_upgrade_test.go:20]: we are skipping cli tests

File: acceptance/tests/connect/connect_inject_test.go
Comment (aahel) [reply acceptance/tests/connect/connect_inject_test.go:216]: have kept these here. since this is something we need to support.  but  test is currently not working.
the ones which are not needed have been skipped and no windows related change are kept

File: acceptance/tests/fixtures/bases/static-client-windows/deployment.yaml
Comment (aahel) [reply acceptance/tests/fixtures/bases/static-client-windows/deployment.yaml:21]: yes we plan to replace these down the line.

File: charts/consul/test/terraform/aks/variables.tf
Comment (aahel) [reply charts/consul/test/terraform/aks/variables.tf:40]: default should be false

File: acceptance/tests/connect/connect_inject_test.go
Comment (Ganeshrockz) [reply acceptance/tests/connect/connect_inject_test.go:216]: May be add a todo at the place you skipped the test.

File: acceptance/framework/config/config.go
Comment (aahel) [reply acceptance/framework/config/config.go]: fixed. https://github.com/hashicorp/consul-k8s/pull/2703/commits/2f2014e8bc64fbeaea93a542031e33dd8fa7949f

File: acceptance/tests/connect/permissive_mtls_test.go
Comment (aahel) [reply acceptance/tests/connect/permissive_mtls_test.go]: fixed https://github.com/hashicorp/consul-k8s/pull/2703/commits/2f2014e8bc64fbeaea93a542031e33dd8fa7949f

File: charts/consul/test/terraform/aks/variables.tf
Comment (aahel) [reply charts/consul/test/terraform/aks/variables.tf:40]: fixed https://github.com/hashicorp/consul-k8s/pull/2703/commits/6ca37eff8b1cf946fde8e8f82d94b50c639c1dff

File: .github/workflows/build.yml
Comment (aahel) [reply .github/workflows/build.yml:11]: will do

File: .github/workflows/build.yml
Comment (aahel) [reply .github/workflows/build.yml]: removed build-windows job https://github.com/hashicorp/consul-k8s/pull/2703/commits/5c3abb022c5c9692cd5a0af7098a39f57050bde8

File: acceptance/framework/config/config.go
Comment (aahel) [reply acceptance/framework/config/config.go:154]: we will sending image name without tag as a test flag and setting tags here to configure windows image or linux image

File: acceptance/framework/connhelper/connect_helper.go
Comment (aahel) [reply acceptance/framework/connhelper/connect_helper.go]: fixed https://github.com/hashicorp/consul-k8s/pull/2703/commits/70221b9e98f14475ae7b87943cec8781a27e4ec5

File: acceptance/tests/api-gateway/api_gateway_external_servers_test.go
Comment (aahel) [reply acceptance/tests/api-gateway/api_gateway_external_servers_test.go:64]: @ishustava 
Can you let us know your opinion on this ?

File: acceptance/framework/k8s/deploy.go
Comment (aahel) [reply acceptance/framework/k8s/deploy.go:72]: @ishustava 
Can you let us know your opinion on this?

File: acceptance/tests/cli/cli_install_test.go
Comment (aahel) [reply acceptance/tests/cli/cli_install_test.go]: unskipped it https://github.com/hashicorp/consul-k8s/pull/2703/commits/b6c16d402c50b42b48669ccb6dba66d13a02e8eb

File: acceptance/tests/cli/cli_upgrade_test.go
Comment (aahel) [reply acceptance/tests/cli/cli_upgrade_test.go]: unskipped it https://github.com/hashicorp/consul-k8s/pull/2703/commits/b6c16d402c50b42b48669ccb6dba66d13a02e8eb

File: acceptance/tests/connect/connect_inject_test.go
Comment (aahel) [reply acceptance/tests/connect/connect_inject_test.go:217]: added. https://github.com/hashicorp/consul-k8s/pull/2703/commits/1bdbf2533357ab5342aa0e0833329643c8f5613e

File: .github/workflows/nightly-acceptance-windows.yml
Comment (aahel) [reply .github/workflows/nightly-acceptance-windows.yml]: fixed. https://github.com/hashicorp/consul-k8s/pull/2703/commits/af9199b5a5eaf5aa819088cba8a2263362260e2f

File: acceptance/framework/config/config.go
Comment (absolutelightning) [reply acceptance/framework/config/config.go:154]: thanks @curtbushko for the clarification

Comment (aahel) [reply acceptance/framework/config/config.go]: have made the change

**Comment by curtbushko** [comment]
@aahel Please squash the commits and get the terraform, unit tests, etc passing.  I will still review with those things broken.

Also, there are most likely acceptance tests fixes in main that need to be rebased into `wa-main-merged` and then into this branch.

**Comment by aahel** [comment]
> @aahel Please squash the commits and get the terraform, unit tests, etc passing. It will make reviewing all the changes easier.
> 
> Also, there are most likely acceptance tests fixes in main that need to be rebased into `wa-main-merged` and then into this branch.

this branch and wa-main-merged is uptodate with main. have fixed tf fmt check. will squash the commits and fix unit tests

**Comment by aahel** [comment]
> > @aahel Please squash the commits and get the terraform, unit tests, etc passing. It will make reviewing all the changes easier.
> > Also, there are most likely acceptance tests fixes in main that need to be rebased into `wa-main-merged` and then into this branch.
> 
> this branch and wa-main-merged is uptodate with main. have fixed tf fmt check. will squash the commits and fix unit tests

@curtbushko unit tests are fixed

**Comment by aahel** [comment]
> > > @aahel Please squash the commits and get the terraform, unit tests, etc passing. It will make reviewing all the changes easier.
> > > Also, there are most likely acceptance tests fixes in main that need to be rebased into `wa-main-merged` and then into this branch.
> > 
> > 
> > this branch and wa-main-merged is uptodate with main. have fixed tf fmt check. will squash the commits and fix unit tests
> 
> @curtbushko unit tests are fixed

@curtbushko do i need to squash all the commits before merge ? coz we do squash and merge right ?

**Pull request closed without merging**
